integration_tests/src/main/python/string_test.py

```diff
@@ -867,6 +867,40 @@ def test_regexp_whitespace():
             ),
         conf=_regexp_conf)
 
+def test_regexp_horizontal_vertical_whitespace():
+    gen = mk_str_gen(
+        '''\xA0\u1680\u180e[abcd]\t\n{1,3} [0-9]\n {1,3}\x0b\t[abcd]\r\f[0-9]{0,10}
+            [\u2001-\u200a]{1,3}\u202f\u205f\u3000\x85\u2028\u2029
+        ''')
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark: unary_op_df(spark, gen).selectExpr(
+                'rlike(a, "\\h{2}")',
+                'rlike(a, "\\v{3}")',
+                'rlike(a, "[abcd]+\\h+[0-9]+")',
+                'rlike(a, "[abcd]+\\v+[0-9]+")',
+                'rlike(a, "\\H{3}")',
```

Comment:
I think this may have the same problems as in https://github.com/NVIDIA/spark-rapids/pull/5479/files#r872460659

We might want to go back and verify that all of the regexp tests in here are doing what we expect them to do.